sormas-backend/src/main/java/de/symeda/sormas/backend/vaccination/VaccinationService.java

```diff
@@ -118,28 +118,28 @@ public List<Vaccination> getVaccinationsByCriteria(
 	}
 
 	public boolean isVaccinationRelevant(Case caze, Vaccination vaccination) {
-		return vaccination.getVaccinationDate() != null
-			&& (caze.getSymptoms().getOnsetDate() != null
-				? DateHelper.getEndOfDay(vaccination.getVaccinationDate()).before(caze.getSymptoms().getOnsetDate())
-				: DateHelper.getEndOfDay(vaccination.getVaccinationDate()).before(caze.getReportDate()));
+		return isVaccinationRelevant(vaccination, caze.getSymptoms().getOnsetDate(), caze.getReportDate());
 	}
 
 	public boolean isVaccinationRelevant(Contact contact, Vaccination vaccination) {
-		return vaccination.getVaccinationDate() != null
-			&& (contact.getLastContactDate() != null
-				? DateHelper.getEndOfDay(vaccination.getVaccinationDate()).before(contact.getLastContactDate())
-				: DateHelper.getEndOfDay(vaccination.getVaccinationDate()).before(contact.getReportDateTime()));
+		return isVaccinationRelevant(vaccination, contact.getLastContactDate(), contact.getReportDateTime());
+
 	}
 
 	public boolean isVaccinationRelevant(Event event, Vaccination vaccination) {
-		if (vaccination.getVaccinationDate() == null) {
-			return false;
-		}
+		Date relevantVaccinationDate =
+			vaccination.getVaccinationDate() != null ? vaccination.getVaccinationDate() : DateHelper.subtractDays(vaccination.getReportDate(), 14);
 		if (event.getStartDate() != null) {
-			return DateHelper.getEndOfDay(vaccination.getVaccinationDate()).before(event.getStartDate());
+			return DateHelper.getEndOfDay(relevantVaccinationDate).before(event.getStartDate());
```

Comment:
you could use ` isVaccinationRelevant(vaccination, event.getStartDate(), null)` so you don't have to duplicate the *relevantVaccinationDate* logic